src/lib/crypto/kimchi_backend/gadgets/xor.ml

```diff
@@ -0,0 +1,333 @@
+open Core_kernel
+
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+module Bignum_bigint = Snarky_backendless.Backend_extended.Bignum_bigint
+
+(* XOR *)
+
+(* Boolean Xor of length bits *)
+let bxor (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    (input1 : Circuit.Field.t) (input2 : Circuit.Field.t) (length : int)
+    (len_xor : int) : Circuit.Field.t =
+  (* Recursively builds Xor *)
+  let rec bxor_rec (input1_bits : bool list) (input2_bits : bool list)
+      (output_bits : bool list) (length : int) (len_xor : int) =
+    let open Circuit in
+    (* Transform to field *)
+    let input1 = Field.Constant.project input1_bits in
+    let input2 = Field.Constant.project input2_bits in
+    let output = Field.Constant.project output_bits in
+    (* Convert to cvar *)
+    let input1_cvar = Common.field_to_cvar_field (module Circuit) input1 in
```

Comment:
Would you want to create a subfunction like so:

`let bits_to_cvar bits = Common.field_to_cvar_field (module Circuit) (Field.Constant.project bits) 
` that could avoid some repetition?